platforms/android/library/src/main/java/io/element/android/wysiwyg/inputhandlers/InterceptInputConnection.kt

```diff
@@ -260,9 +274,10 @@ internal class InterceptInputConnection(
                 processInput(action)
             }
             if (result != null) {
-                replaceAll(result.text, result.selection.first, result.selection.last)
-                setSelectionOnEditable(editable, result.selection.first, result.selection.last)
-                setComposingRegion(result.selection.first, result.selection.last)
+                replaceAll(result.text)
+                val editorSelectionIndex = editorIndex(result.selection.first, editable)
+                setSelection(editorSelectionIndex, editorSelectionIndex)
+                setComposingRegion(editorSelectionIndex, editorSelectionIndex)
```

Comment:
nit: I think it's reasonable to assume that `selection.first == selection.last` but it feels a bit strange not to trust the Rust model. Anyway, it's not a big deal but if we did manage to refactor the result handling into a function (as mentioned in my other comment), that might fix this.

Reply:
I mostly did it following what was already there and trying to avoid calculating indexes several times, I see no need to convert twice index `x` from the composer to get the same `y` editor index in both cases, doing extra work. Also, the composer doesn't have the concept of 'composition', so we some times may have to get a bit creative with selections.

In the future we could add this concept to the composer, since it seems to be used in both Android an iOS (on iOS I believe it's called 'marked text'), but right now that would be way too much work for the time we have to work on the library.